### ISSUE TITLE
Validation Exception when creating Environment

### ISSUE THREAD

**Issue by ttirtawi** [mreferre/aws-proton-tutorial#3]
Got the following error when creating the Environment (with create new role option)

```
Validation exception
Must supply one and only one of: arrow service role arn, environment account connection id, or provisioning repository arn.
```

**Comment by ttirtawi** [comment]
Seems there is bug on the IAM service role creation.When I check the role is created but it has empty policy. It later cause the issue when continue to create & delete.

```
Provided AWS Proton service role: arn:aws:iam::9160497xxxxx:role/service-role/test-proton-env-role is not authorized to perform: cloudformation:DescribeStacks on resource: arn:aws:cloudformation:ap-northeast-1:9160497xxxxx:stack/AWSProton-Dev-cloudformation--FURSWHTOKDTIDWY/* because no identity-based policy allows the cloudformation:DescribeStacks action
```

**Comment by mreferre** [comment]
Thanks for reporting this! Mh... can you describe the exact steps to recreate the problem? I assume you are doing everything from the console?

**Comment by ttirtawi** [comment]
Yes, I follow your tutorial and do everything from console. Issue when creating the Environment, choose to create new role, then get the validation exception issue. After few trial, it can be created but the service role created without no policies attached.

**Comment by mreferre** [comment]
Oh I think I see what's going on. Are you by any chance creating the environment using a user that is not a full account administrator (e.g. a user with the `AWSProtonFullAccess` policy?). I have tried with that user and I get the same error (see below). The reason is that the role this wizard tries to create has full admin capabilities and your user has not enough privileges to do that. The `AWSProtonFullAccess` policy can't create an Administrator role but, right now, it misses a few capabilities to be able to list a role pre-created (we need to fix this). 

As a work around, can you complete the tutorial with an Administrator user until we fix this catch-22? Note to self: I need to remove the `As a stretch goal, if you want, you may want to create two separate users with two separate AWS managed IAM policies (AWSProtonFullAccess and AWSProtonDeveloperAccess)` text in the tutorial! 

![image](https://user-images.githubusercontent.com/4123965/156893424-a9cecbe0-a749-4614-8cf0-5c4b2ec7e37d.png)

**Comment by ttirtawi** [comment]
Unfortunately I did the tutorial using user that already use `AdministratorAccess` policy.

**Comment by mreferre** [comment]
Mh.... that's weird. I have had multiple users going through the tutorial (with an Administrator level access) and they all got past this point. Do you mind sending me an email with the region + account number you are using? (mreferre at amazon.com). Thanks!

**Comment by mreferre** [comment]
Update: I am currently investigating this issue. If you are doing the tutorial and you are reading this, as a workaround please try to create the user out of band with the the `AdministratorAccess` policy and this trust relationship:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Service": "proton.amazonaws.com"
            },
            "Action": "sts:AssumeRole"
        }
    ]
}
```

**Comment by mreferre** [comment]
We have confirmed this seems to be a problem with the console workflow that creates the role. Apparently the role is not fully in place before the Proton API is called thus the failure (if you re-run the workflow and picking the newly created role instead of creating a new one it will work).

**Comment by ghost** [comment]
The issue is that we create the role and policies on the fly and it may take up to 2 mins before you can use these resources though most times it's done within seconds. We're are working on a fix now.
